cognite/neat/session/_plugin.py

```diff
@@ -0,0 +1,65 @@
+import warnings
+from typing import Any
+
+from cognite.neat.core._issues._base import IssueList
+from cognite.neat.core._utils.reader._base import NeatReader
+from cognite.neat.plugins import manager as plugin_manager
+from cognite.neat.plugins.data_model.importers._base import DataModelImporterPlugin
+from cognite.neat.session._experimental import ExperimentalFlags
+
+from ._state import SessionState
+from .exceptions import session_class_wrapper
+
+
+@session_class_wrapper
+class PluginAPI:
+    """Read from a data source into NeatSession graph store."""
+
+    def __init__(self, state: SessionState) -> None:
+        self._state = state
+
+        self.data_model = DataModelPlugins(self._state)
+
+
+@session_class_wrapper
+class DataModelPlugins:
+    """Read from a data source into NeatSession graph store."""
+
+    def __init__(self, state: SessionState) -> None:
+        self._state = state
+
+    def read(self, name: str, io: Any | None = None, **kwargs: Any) -> IssueList:
+        """Provides access to the external plugins for data model importing.
+
+        Args:
+            name (str): The name of format (e.g. Excel) plugin is handling.
+            io (Any | None): The input/output interface for the plugin.
+            **kwargs (Any): Additional keyword arguments for the plugin.
+
+        !!! note "kwargs"
+            Users must consult the documentation of the plugin to understand
+            what keyword arguments are supported.
+        """
+        warnings.filterwarnings("default")
+        ExperimentalFlags.plugin.warn()
+
+        # Some plugins may not support the io argument
+        if io:
+            reader = NeatReader.create(io)
+            path = reader.materialize_path()
+        else:
+            path = None
+
+        self._state._raise_exception_if_condition_not_met(
+            "Data Model Read",
+            empty_data_model_store_required=True,
+        )
+
+        plugin = plugin_manager.get(name, DataModelImporterPlugin)
+
+        print(
+            f"You are using an external plugin {plugin.__name__}, which is not developed by the NEAT team."
+            "\nUse it at your own risk."
+        )
+
+        return self._state.data_model_import(plugin().configure(io=path, **kwargs))
```

Comment:
`configure` also has the same issue, effectively. At this point `io` is `str | Path`, and `configure should probably reflect this.

I'd recommend you define a `NeatIO` type alias, or something like that, since this type appears to be used quite widely.